### ISSUE TITLE
[Bug] \Phalcon\Mvc\Dispatcher does not set Last Handler on Exception

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#11174]
When an exception occurs dispatching, the last handler will not get set.  The setting of the last handler is only performed on success but needs to happen in **all** cases as subsequent error handling actions further up the stack that may dispatch again would require this information (as in the case in one of my error handling applications) This could easily be achieved with a finally block for the try/catch and moving the setting of the last handler into that section.

See
https://github.com/phalcon/cphalcon/blob/master/phalcon/dispatcher.zep#L510

**Comment by rianorie** [comment]
Can you produce a script that reproduces the behavior?

**Comment by virgofx** [comment]
`IndexController.php`
```php
class IndexController extends \Phalcon\Mvc\Controller
{
    public function indexAction() {
        throw new \MyCustomException('custom exception');
    }
}
```

`bootstrap.php`
```php
$di = new \Phalcon\Di\FactoryDefault();
try {
		$application = new \Phalcon\Application();
		$application->setEventsManager(new \Phalcon\Events\Manager()));
		$application->setDI($di);

		echo $application->handle()->getContent();

	} catch (\MyCustomException $e) {
		// 'IndexController'  [OK]
		var_dump($di->getDispatcher()->getActiveController());
		
		// [empty]  [ WRONG ]
		// [Expected: "IndexController"]
		var_dump($di->getDispatcher()->getLastController());

		// Custom logic here to send errors or do something custom that requires
		// knowledge of the last controller.
	}
}
```

**The Fix**

_Change_
```php
try {
	// We update the latest value produced by the latest handler
	let this->_returnedValue = call_user_func_array([handler, actionMethod], params),
		this->_lastHandler = handler;

} catch \Exception, e {
	if this->{"_handleException"}(e) === false {
		if this->_finished === false {
			continue;
		}
	} else {
		throw e;
	}
}
```


_To_
```php
try {
	// We update the latest value produced by the latest handler
	let this->_returnedValue = call_user_func_array([handler, actionMethod], params);

} catch \Exception, e {
	if this->{"_handleException"}(e) === false {
		if this->_finished === false {
			continue;
		}
	} else {
		throw e;
	}
} finally {
	this->_lastHandler = handler;
}
```

**Comment by rianorie** [comment]
.. you've adjusted the zephir code as such? Then why not make it a pull request? :)

I'm happy to do it, though

**Comment by virgofx** [comment]
I've extended in PHP land for my projects, which do essentially above ^^. I currently don't have Zephir checked out , which is why I haven't done it yet. If you could add the fix, that'd be great ;)

**Comment by sergeyklay** [comment]
Fixed in `2.0.x`
